CMakeLists.txt

```diff
@@ -302,50 +331,102 @@ endif()
 
 # Link Options
 # Find FFMpeg stuff
-find_library(LIB_AVCODEC libavcodec HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
-find_library(LIB_AVDEVICE libavdevice HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
-find_library(LIB_AVFILTER libavfilter HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
-find_library(LIB_AVFORMAT libavformat HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
-find_library(LIB_AVUTIL libavutil HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
-find_library(LIB_SWRESAMPLE libswresample HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
-find_library(LIB_SWSCALE libswscale HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
-
-target_link_libraries(
-    MathAnimations 
-    glfw
-    freetype
-    DearImGuiBackend
-    Glad
-    NativeFileDialog
-    OpenAL
-    Luau.Ast 
-    Luau.CodeGen 
-    Luau.Compiler 
-    Luau.VM 
-    Luau.Analysis 
-    onig 
-    plutovg 
-    tinyxml2 
-    DearImGui
-    # FFmpeg stuff
-    ${LIB_AVCODEC}
-    ${LIB_AVDEVICE}
-    ${LIB_AVFILTER}
-    ${LIB_AVFORMAT}
-    ${LIB_AVUTIL}
-    ${LIB_SWRESAMPLE}
-    ${LIB_SWSCALE}
-    # Windows static libs required for ffmepg
-    Ws2_32
-    Secur32
-    Bcrypt
-    Mfuuid
-    Strmiids
-)
+if (MSVC)
+    find_library(LIB_AVCODEC libavcodec HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
+    find_library(LIB_AVDEVICE libavdevice HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
+    find_library(LIB_AVFILTER libavfilter HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
+    find_library(LIB_AVFORMAT libavformat HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
+    find_library(LIB_AVUTIL libavutil HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
+    find_library(LIB_SWRESAMPLE libswresample HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
+    find_library(LIB_SWSCALE libswscale HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
+else()
+    find_library(LIB_AVCODEC avcodec HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
```

Comment:
Is there any reason we have this `if-else` check? It looks identical to me but I may be missing the change here.

Reply:
The Unix variant is missing the `lib` prefix for the library names, which I just assumed was a thing MSVC needed.
It might also not be necessary with MSVC/WIN32, would be cool if that could be checked.

Reply:
I'll double check but I think CMake automatically appends the `.lib` on Windows :)

Reply:
Not append .lib, but prepend lib
The output files are e.g. libavcodec.a/.lib, but gcc and clang expect -lavcodec instead of -llibavcodec

Reply:
Oh I see now, I wasn't seeing the `lib` in front of the libraries haha